### PR TITLE
Refactor handling of java opts env vars in cli

### PR DESCRIPTION
**Pull request by Xenorith**
address https://github.com/Alluxio/alluxio/issues/18249

refactor how java opt env vars are handled. previously they were treating as strings and then using a split operation to separate the single string into cmd line arguments. now they are handled as `[]string` immediately after parsing from env to properly handle any leading or trailing whitespaces.

### CODE REVIEW

**Comment by Xenorith** [comment]
may be easier to review individual commits as there are many no-op changes

**Comment by Xenorith** [comment]
also even though there is a unit test added, it is not currently running as part of the PR build. i will need to add additional logic to run it in a separate PR.

but the unit test did catch a few cases i forgot to handle, so i feel pretty confident about the change now

File: cli/src/alluxio.org/cli/env/env_vars.go [hunk 1]
```diff
@@ -29,6 +30,7 @@ type AlluxioConfigEnvVar struct {
 	description string // describes the environment variable in conf/alluxio-env.sh.template file
 
 	// optional
+	IsJavaOpts                bool              // true if environment variable's value is a java opts string
```
Comment (Kai-Zhang) [inline cli/src/alluxio.org/cli/env/env_vars.go:33]: i didn't find any usage of this bool. why is this necessary?

**Review by Kai-Zhang — APPROVED** [review]
mostly LGTM

File: cli/src/alluxio.org/cli/env/env_vars.go (hunk 1 above)
Comment (Xenorith) [reply cli/src/alluxio.org/cli/env/env_vars.go:33]: agh i forgot to use it x.x

**Comment by Xenorith** [comment]
alluxio-bot, merge this please
